### PR TITLE
Update profiling section

### PR DESCRIPTION
**Pull request by fsimonis**
This PR updates the profiling section according to https://github.com/precice/precice/pull/1419 

TODO:

* [x] recreate output after the preCICE PR is merged
* [x] add example of a supemuc run
* [x] add comparison outputs between all and fundamental modes

### CODE REVIEW

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
These named sections are called `events` and each event generates records during runtime.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md:10]: It could be good to have one sentence somewhere on why we develop our own profiling framework.

> Existing external frameworks cannot cope with the complexity of the multi-executable setup of preCICE. To understand the performance of a coupled run, it is necessary to look at the interplay of all coupled participants.

or similar

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
Each rank of each participant gathers local records and writes them to a JSON file. To store these JSON files, preCICE creates a directory called `precice-events` in the current working directory of each participant.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
All these files of all participants then need to be post-processed and merged to a single file using the `precice-events` python script.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
To keep this feature as useful as possible, we pre-selected a set of events that we deem fundamental.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
These include the main preCICE API functions and the time spent between these calls inside the solvers.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
* `SolverInterface` construction. This includes configuration and setting up the intra-communication of each participant.
```
?

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
* `solver.initialize` time spent in the solver until `initialize()` is called. This typically includes setting up meshes, defining initial data and preparing the solver.
* `initialize()` time spent in preCICE `initialize()`. This includes establishing communication between participants, mesh and data transfer, as well as mapping computation.
* `solver.advance` time spent in the solver between `advance()` calls, including the time between `initialize()` and the first `advance()` call.
* `advance()` time spent in preCICE `advance()`. This includes data mapping, data transfer, acceleration.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
Measuring the runtime of such operations can be tricky, as the time spent waiting should not be misinterpreted.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
Synchronizing all ranks using a barrier solves the issue, but has an impact on performance.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]: This, we still want to change. Then, it could make sense to first introduce the profiling tag.

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
You can configure the profiling with the `<profiling/>` tag located inside `<precice-configuration>`.
```
The last part will need updating.

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
* the flush frequency  `flush-every`: `0` means only at the end of the simulation, `n>0` means to flush every n records
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
To write the files of all participants into a common directory (often `..` works too):
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md:93]: Will need updating

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
3. Analyze each participants that executes a mapping
4. Analyze each participant to check for load-imbalance between ranks
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]: Why the focus on mappings?
Mesh repartitioning, acceleration, ... could also get important.

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
After the simulation completes, you can find `precice-events` folders in the configured location, defaulting to the working directory of each participant.
```
Just to stick to the same name here.

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md:139]:
```suggestion
Writing to events.json
$ ls
A
B
events.json
```
?

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
You can also pass individual files if you are not interested in all ranks
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
### Visualizing the simulation
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
An example trace visualization using `ui.perfetto.dev` of a serial [ASTE](tooling-aste.html) run looks as follows:
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
An example trace visualization using `ui.perfetto.dev` of a parallel [ASTE](tooling-aste.html) run on two ranks each looks as follows:
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
### Analyzing participants
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
After the name of the event, the table contains three blocks, each containing the sum, count, mean, min, and max runtime for a specific rank.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
1. The first block displays the primary rank (0).
2. The second block displays the secondary rank which spent the least total time in `advance`.
3. The third block displays the secondary rank which spent the most total time in `advance`.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]: Cheapest and most expensive ranks are identical in this example.

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
This contains CSV data including all individual events from all participants and ranks.
```

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]:
```suggestion
It also contains additional data entries attached to events.
```
Could we give an example here?

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (uekerman) [inline pages/docs/tooling/tooling-performance-analysis.md]: Currently not true. I would suggest to simplify the process above.

**Review by uekerman — APPROVED** [review]
Very nice job :+1:
No need for a second review iteration.
Don't forget to update this page as well: https://precice.org/running-output-files.html

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (fsimonis) [reply pages/docs/tooling/tooling-performance-analysis.md]: Configuring a mapping often implies repartitioning. I added acceleration though.

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (fsimonis) [reply pages/docs/tooling/tooling-performance-analysis.md]: Replaced it with an ASTE run

File: pages/docs/tooling/tooling-performance-analysis.md
Comment (fsimonis) [reply pages/docs/tooling/tooling-performance-analysis.md]: I reworded this to be more general.

**Comment by fsimonis** [comment]
Last thing to do is to implement https://github.com/precice/precice/issues/1634 and update the configs.

This will only change the two configs. So this is ready for an additional review.

**Comment by uekerman** [comment]
> Don't forget to update this page as well: https://precice.org/running-output-files.html

:wave: :grin:

**Comment by fsimonis** [comment]
I'll merge this in the develop branch. Let's iterate on feedback afterwards

**Comment by MakisH** [comment]
Looks really nice! I have fixed some minor issues in https://github.com/precice/precice.github.io/pull/272. Sorry for the late review.

**Pull request merged**
